File: src/main/java/me/oriharel/machinery/api/events/PreMachineBuildEvent.java
package me.oriharel.machinery.api.events;

import me.oriharel.machinery.machine.Machine;
import org.bukkit.Location;
import org.bukkit.event.Cancellable;
import org.bukkit.event.HandlerList;

public class PreMachineBuildEvent extends MachineEvent<Machine> implements Cancellable {
    private static final HandlerList HANDLERS = new HandlerList();

    private boolean cancelled;

    public PreMachineBuildEvent(Machine machine, Location loc) {
        super(machine, loc);
    }

    public static HandlerList getHandlerList() {
        return HANDLERS;
    }

    public HandlerList getHandlers() {
        return HANDLERS;
    }

    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    @Override
    public void setCancelled(boolean cancel) {
        this.cancelled = cancel;
    }
}

File: src/main/kotlin/me/oriharel/machinery/machines/events/PreMachineBuildEvent.kt
package me.oriharel.machinery.machines.events

import me.oriharel.machinery.machines.machine.Machine
import org.bukkit.Location
import org.bukkit.event.Cancellable
import org.bukkit.event.HandlerList

class PreMachineBuildEvent(machine: Machine?, loc: Location) : MachineEvent<Machine?>(machine, loc), Cancellable {
    private var cancelled = false
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    override fun isCancelled(): Boolean {
        return cancelled
    }

    override fun setCancelled(cancel: Boolean) {
        cancelled = cancel
    }

    companion object {
        val handlerList = HandlerList()
    }
}